### PR TITLE
CIGI-527: Implement newsletter page

### PR DESCRIPTION
**Pull request by johnbxu**
#### Description of changes
resolves CIGIHub/cigi-tickets#527
resolves CIGIHub/cigi-tickets#546

http://127.0.0.1:8000/newsletters/angie-drobnic-holan-importance-fact-checking-during-pandemic/
http://127.0.0.1:8000/newsletters/deliberate-disinformation-and-global-health-security/

image

image

### CODE REVIEW

**Review by gabrieltanchen — COMMENTED** [review]
I'm a little confused as to what's going on with the images. The URL is hardcoded for production but it's using Wagtail image paths?

**Comment by johnbxu** [comment]
> The URL is hardcoded for production but it's using Wagtail image paths?

The Wagtail image path is relative, but the html needs to use the production path because it's the one copied to mailchimp. On the Drupal site there is a function in the theme hook to replace all admin urls with non-admin. I could do that instead?

**Comment by gabrieltanchen** [comment]
Oh okay I see where you're hardcoding the domain, and I assume this is why you asked if editors will still be accessing from admin.cigionline.org?

You should be able to use the domain of the root site here. Even for admin.cigionline.org, it'll still be accessing the production database which will have the root site as www.cigionline.org.

**Comment by johnbxu** [comment]
Ah ok, good to know.

**Comment by johnbxu** [comment]
Updated to use root domain

File: streams/blocks.py [hunk 1]
```diff
@@ -631,3 +632,180 @@ class Meta:
...
+            context['title'] = value.get('title_override') if value.get('title_override') else content_page.title
+            context['text'] = value.get('text_override') if value.get('text_override') else content_page.specific.short_description
+            if value.get('image_override'):
+                context['image_url'] = value.get("image_override").get_rendition("fill-600x238").url
+            elif content_page.specific.image_hero:
+                context['image_url'] = content_page.specific.image_hero.get_rendition("fill-600x238").url
+                context['image_alt'] = content_page.specific.image_hero.title
+
+            if not value.get('url'):
+                context['url'] = f'https://www.cigionline.org{content_page.url_path}'
```
Comment (gabrieltanchen) [inline streams/blocks.py:712]: Beware of using `url_path` in links, since it contains the home slug as well. Can this also check for the root domain?

**Review by gabrieltanchen — CHANGES_REQUESTED** [review]
The site in the request doesn't seem to be the default site, I believe it's returning `None` since the request domain doesn't match. You should be able to query for the default Site in the `get_context` method, then use the `root_url` property: https://docs.wagtail.io/en/stable/reference/pages/model_reference.html#wagtail.core.models.Site.root_url

Using the `image_url` or image object should be cleaned up between the `get_context` method and the block templates.

**Comment by johnbxu** [comment]
Used site root_url from context

**Comment by gabrieltanchen** [comment]
@johnbxu plot twist on this. After fixing some stuff with the social meta tags, I don't think we need `root_url` here at all. Just let the image urls get generated as normal, and they should pull from the root domain automatically because of the S3 settings on production.

Can you fix this to remove the root url code? If we run into any issues when setting up admin we can fix it then.

**Comment by johnbxu** [comment]
Removed root_url from images, and removed social links as per https://github.com/CIGIHub/cigi-tickets/issues/546

**Review by gabrieltanchen — APPROVED** [review]

**Pull request merged**